lib/challenge.js

```diff
@@ -51,16 +78,38 @@ var pet_owner;
 var some_pet;
 
 // Answer Starts Here
-
+var some_pet = {
+  name: "Bianca",
+  species: "K9",
+  breed: "Bernese Mountain Dog",
+  noise: "Woof"
+}
 // Answer Ends Here
 
 // 4. Create a constructor function for Pet objects, as described above. The constructor function should be responseible for setting
 //    the values of `name`, `species`, `breed`, and `noise`. The constructor function should also give each pet a method called `makeNoise`,
 //    which behaves as described in Q3.
 
-var Pet;
+var Pet ;
 // Answer Starts Here
-
+var Pet = function Pet (name,species,breed,noise){
+    this.name = name;
+    this.species = species;
+    this.breed = breed;
+    this.noise = [];
+}
+Pet.prototype.makeNoise = function(species){
+    var noise = 0;
+    for( var i = 0; i < this.Pet.length; i++){
+      if(species === 'K9'){
+        noise = "Woof";
+      }
+      else {
+        noise = "Meow";
+      }
+    }
+    return noise;
+}
```

Comment:
The prototype piece of this is right, but the `makeNoise` method should reference the `noise` property.
e.g.
```javascript
Pet.prototype.makeNoise = function(species){
  return this.noise;
}
```